### PR TITLE
Remove unnecessary toString conversion of vector field and added some minor optimization in KNNCodec

### LINKED ISSUE #1598: [Enhancement][Build-Time] Remove conversion of KNN Field to string if stored Fields not required
### Description
While doing some benchmarks with Opensearch k-NN. I see that there is time going in to convert the kNN field to String even when storedFields is marked as false. On doing deep-dive I found out that, we convert the field to the string and then check if stored field is set as true or false.

Code ref:
https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java#L575

Here `point.toString` will convert the object to string which is expensive operation in high dimensionality use cases. 

Refer the Flame graph for the same for Vector dimension: 1536 ingestion

Screenshot 2024-04-08 at 12 58 58 PM

### PR DESCRIPTION
**Pull request by navneet1v**
### Description
Remove unnecessary toString conversion of vector field and added some minor optimization in KNNCodec
 
### Issues Resolved
#1598
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ryanbogan — APPROVED** [review]
LGTM!

File: src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperUtil.java [hunk 1]
```diff
@@ -135,14 +136,9 @@ public static FieldType buildDocValuesFieldType(KNNEngine knnEngine) {
...
-    public static void addStoredFieldForVectorField(
-        ParseContext context,
-        FieldType fieldType,
-        String mapperName,
-        String vectorFieldAsString
-    ) {
+    public static void addStoredFieldForVectorField(ParseContext context, FieldType fieldType, String mapperName, Field vectorField) {
         if (fieldType.stored()) {
-            context.doc().add(new StoredField(mapperName, vectorFieldAsString));
+            context.doc().add(new StoredField(mapperName, vectorField.toString()));
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperUtil.java:141]: so the idea is - we only covert to String if field is stored, only always. That's good optimization!

**Review by martin-gaievski — APPROVED** [review]
looks good, thank you

**Review by junqiu-lei — APPROVED** [review]

File: src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperUtil.java (hunk 1 above)
Comment (navneet1v) [reply src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperUtil.java:141]: Yes that is correct and if you look into the issue you can see the time it takes to convert a field to string.

**Pull request merged**